src/Internal/Payment/PaymentMethod/SavedPaymentMethod.php

```diff
@@ -7,26 +7,33 @@
 
 namespace WCPay\Internal\Payment\PaymentMethod;
 
-use WC_Payment_Token;
-
 /**
- * Representation of a newly entered payment method.
+ * Representation of a saved payment method.
  */
 class SavedPaymentMethod implements PaymentMethodInterface {
 	/**
-	 * Contains the WooCommerce token.
+	 * External ID of the payment method.
+	 *
+	 * @var string
+	 */
+	private $id;
+
+	/**
+	 * Contains the WooCommerce token ID.
 	 *
-	 * @var WC_Payment_Token
+	 * @var int
 	 */
-	private $token;
+	private $token_id;
 
 	/**
 	 * Class constructor.
 	 *
-	 * @param WC_Payment_Token $token The WooCommerce token.
+	 * @param string $id       External ID of the payment method.
+	 * @param int    $token_id Internal WooCommerce token ID.
 	 */
-	public function __construct( WC_Payment_Token $token ) {
-		$this->token = $token;
+	public function __construct( string $id, int $token_id ) {
+		$this->id       = $id;
+		$this->token_id = $token_id;
 	}
```

Comment:
I understand we'd like to simplify `SavedPaymentMethod` so that it can be easily serializable in `PaymentContext`. 

But these construct parameters lose the tight contract with `WC_Payment_Token`. IMO, we should still `WC_Payment_Token` as a single contract parameter, but does not save it within the `SavedPaymentMethod` object.

```suggestion
	public function __construct( \WC_Payment_Token $woo_token ) {
		$this->id       = $woo_token->get_id();
		$this->token_id = $woo_token->get_token();
	}
```

Reply:
Please see https://github.com/Automattic/woocommerce-payments/pull/7324#discussion_r1344193946. The reason to avoid carrying the `WC_Token` object around is the same as orders, and interact through a token (could be called something else) service. It seemed a bit weird at first, but realistically we don't do much with tokens throughout the payment process. As long as I can call `$context->get_payment_method()->get_id()`, I'm happy. :)

Reply:
Agreed that we would avoid carrying Woo core objects. However, in my suggestion, we're NOT carrying `WC_Payment_Token`, we just simply use it as a construct parameter to populate the private properties for `SavedPaymentMethod`.

Reply:
I have considered this option, the problem is that we would require the token service every time we want to re-construct the object because we'd need to load a `WC_Payment_Token` object. AFAIK, tokens would not change: I'm not aware of any flows, which would change the token within a `WC_Payment_Token` object. With that in mind, do you think that there is an upside to having the object?

Reply:
> the problem is that we would require the token service every time we want to re-construct the object because we'd need to load a `WC_Payment_Token` object.

It's the problem if we want to re-construct the `SavedPaymentMethod` object often. But I think when it's attached to `PaymentContext`, we mostly only care `$context->get_payment_method()->get_id()` (similar to what you mentioned above). 

Anyway, this is just a suggestion rather than a blocker. I also agree that we do not often instantiate `SavedPaymentMethod`.

Reply:
> `SavedPaymentMethod::from_wc_payment_token` which could also perform any checks if the token is valid.

What would it validate, and as a static method, could it do it without a dependency?

Reply:
We could, but we have not done anything like this so far, and I think we're on the safe side. Keep in mind that the `PaymentRequest` class already checks if the gateway is correct, and I wouldn't assume an empty token can be stored.

I was thinking that we'd need to validate whether the token still exists, but that will happen during reconstruction anyway. If we get to a point with duplicate code, we could always create the static method. Presently, it seems not needed, right?.